src/main/java/ch/wisv/areafiftylan/service/SeatServiceImpl.java

```diff
@@ -1,43 +1,124 @@
+
+        for (Seat seat : all) {
+            if (seatMapResponse.containsKey(seat.getSeatGroup())) {
+                seatMapResponse.get(seat.getSeatGroup()).add(seat);
+            } else {
+                List<Seat> seats = new ArrayList<>();
+                seats.add(seat);
+                seatMapResponse.put(seat.getSeatGroup(), seats);
+            }
+        }
+
+        return new SeatmapResponse(seatMapResponse);
+    }
+
+    @Override
+    public SeatmapResponse getSeatGroupByName(String groupname) {
+        List<Seat> seatGroup = seatRespository.findBySeatGroup(groupname);
+
+        Map<String, List<Seat>> seatMapResponse = new HashMap<>();
+        seatMapResponse.put(groupname, seatGroup);
+
+        return new SeatmapResponse(seatMapResponse);
+    }
+
+    @Override
+    public boolean reserveSeatForTicket(String groupname, int seatnumber, Long ticketId) {
+        Ticket ticket = ticketRepository.findOne(ticketId);
+        Seat seat = seatRespository.findBySeatGroupAndSeatNumber(groupname, seatnumber);
+        if (!seat.isTaken()) {
+            seat.setTicket(ticket);
+            seatRespository.saveAndFlush(seat);
+            return true;
+        } else {
+            return false;
+        }
     }
 
     @Override
-    public Seat getSeatByCoordinate(Coordinate coordinate) {
-        return seatRepository.findByCoordinate(coordinate);
+    public boolean reserveSeat(String groupname, int seatnumber) {
+        Seat seat = seatRespository.findBySeatGroupAndSeatNumber(groupname, seatnumber);
+
+        if (!seat.isTaken()) {
+            seat.setTaken(true);
+            seatRespository.saveAndFlush(seat);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public Seat getSeatBySeatGroupAndSeatNumber(String groupName, int seatNumber) {
+        return seatRespository.findBySeatGroupAndSeatNumber(groupName, seatNumber);
     }
 
     @Override
-    public Seat reserveSeat(Coordinate coordinate, User user) {
-        Seat seat = seatRepository.findByCoordinate(coordinate);
-        seat.setUser(user);
-        return seatRepository.saveAndFlush(seat);
+    public void addSeats(SeatGroupDTO seatGroupDTO) {
+        List<Seat> seatList = new ArrayList<>(seatGroupDTO.getNumberOfSeats());
+
+        for (int i = 1; i <= seatGroupDTO.getNumberOfSeats(); i++) {
+            seatList.add(new Seat(seatGroupDTO.getSeatGroupName(), i));
+        }
+
+        seatRespository.save(seatList);
+    }
+
+    @Override
+    public Set<Seat> getSeatsByTeamName(String teamName) {
+        Set<Seat> seats = new HashSet<>();
+        Team team = teamService.getTeamByTeamname(teamName).orElseThrow(() -> new TeamNotFoundException(teamName));
+
+        for (User user : team.getMembers()) {
+            Optional<Seat> seat = seatRespository.findByTicketOwnerUsername(user.getUsername());
+            if (seat.isPresent()) {
+                seats.add(seat.get());
```

Comment:
I think these lines can be replaced with

```java
seatRespository.findByTicketOwnerUsername(user.getUsername()).ifPresent(seats::add);
```

See: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-